nifi-nar-bundles/nifi-standard-services/nifi-ssl-context-bundle/nifi-ssl-context-service/src/main/java/org/apache/nifi/ssl/StandardSSLContextService.java

```diff
@@ -582,6 +598,6 @@ public String toString() {
             allowableValues.add(new AllowableValue(supportedProtocol, supportedProtocol, description));
         }
 
-        return allowableValues.toArray(new AllowableValue[allowableValues.size()]);
+        return allowableValues.toArray(new AllowableValue[0]);
```

Comment:
Is there a reason for changing the AllowableValue array size to 0 here?

Reply:
Performance and memory

The java api here simply needs to know what object type to use for the array being constructed, it doesn't use the array that's created for the parameter in the `toArray` method)... so simply using 0 here means we save the call to `.size` and memory of allocating an unnecessarily large array that's unused

Reply:
Thanks for the helpful reply!